### ISSUE TITLE
oxfmt: --default() inside --value() in a Tailwind @utility triggers spurious 'Unsupported bare value data type' warnings

### ISSUE THREAD

**Issue by benface** [oxc-project/oxc#26187]
### Summary

`oxfmt --check` prints `Unsupported bare value data type: "-*"` warnings for every Tailwind `@utility name-*` block whose declarations use `--default(...)` inside `--value(...)`. Real Tailwind (`@tailwindcss/node`, tested on 4.3.3) compiles the exact same CSS without warnings — the diagnostic appears to originate in oxfmt's own Tailwind-classes sorter (`sortTailwindcss`), and the caret in the message points at `-*` characters that don't exist in the source.

### Version

- `oxfmt` 0.65.0
- Node 26.7.0
- Tailwind 4.3.3 (via `sortTailwindcss.stylesheet`)

### Minimal repro

`utils.css`:

```css
@utility rule-x-* {
  --w: --value([*], --default(1px));
}
```

`entry.css`:

```css
@import "tailwindcss";
@import "./utils.css";
```

`.oxfmtrc.json`:

```json
{ "sortTailwindcss": { "stylesheet": "./entry.css" } }
```

Run:

```
oxfmt --check .
```

### Actual output

```
Unsupported bare value data type: "-*".
Only valid data types are: "number", "integer", "ratio", "percentage".

```css
--value([*],--default(1px)-*)
                          ^^
```
```

The trailing `-*` in the "context" isn't in the source — it looks like the `-*` suffix from the `@utility rule-x-*` name is being concatenated onto the last `--value(...)` expression when producing the diagnostic.

### Expected

No warning: `--default(...)` is a valid Tailwind v4 argument to `--value()` and `--modifier()` (see [Tailwind docs — Default values](https://tailwindcss.com/docs/adding-custom-styles#default-values)). Compiling the same CSS with `@tailwindcss/node`'s `compile()` produces correct output for `.rule-x`, `.rule-x-2`, and `.rule-x-[3px]` with zero warnings.

### Workaround

Replace `--default(...)` with a plain `var(--fallback, 1px)` — semantically different (loses the naked-utility default behavior), but silences the warnings.

### Impact

Every `@utility name-*` block using `--default(...)` emits one warning per Tailwind-sorter load, cluttering CI logs (24 warnings from a single small file in our repo, across two projects that reference the same `monowind` package). Exit code is 0, so it's cosmetic — but easy to miss real warnings underneath.

**Comment by benface** [comment]
### Root cause pinned

This isn't an oxfmt-side bug — it's a **stale bundled Tailwind** inside `prettier-plugin-tailwindcss`, which oxfmt pins as a dependency.

**The exact missing guard** — in `--value(…)` argument normalization inside Tailwind's `utilities.ts`:

Real Tailwind 4.3.3 (minified, unaffected):
```js
p[0]==="-"&&p[1]==="-"&&!p.includes("(")&&!p.includes("-*")&&(p+="-*")
```

Bundled inside `oxfmt@0.65.0`'s `dist/v4-C-HWEQJm-BVYBnYoU.js` (missing `!p.includes("(")`):
```js
f[0] === "-" && f[1] === "-" && !f.includes("-*") && (f += "-*")
```

Without the `!p.includes("(")` guard, `--default(1px)` gets `-*` appended → becomes `--default(1px)-*` → the value parser then sees a bare word `-*` and warns.

**Upstream chain:**

1. Guard was added in Tailwind 4.3.0 ([tailwindlabs/tailwindcss@08cad84](https://github.com/tailwindlabs/tailwindcss/commit/08cad84), PR [tailwindlabs/tailwindcss#19989](https://github.com/tailwindlabs/tailwindcss/pull/19989), 2026-05-08).
2. `prettier-plugin-tailwindcss@0.0.0-insiders.fad3e07` (2026-07-15) pins `tailwindcss@^4.1.14` — a fresh install today resolves to 4.3.3, but the published insider bundle still ships pre-4.3.0 code. Confirmed by diffing the two `v4-*.mjs` bundles: only the wrapping import paths differ.
3. `oxfmt@0.65.0` pins `prettier-plugin-tailwindcss: "0.0.0-insiders.2b6f3c2"` (2026-05-25), which has the same stale code.

**Suggested fix path:** bump `prettier-plugin-tailwindcss` (either to a rebuild of the insider that vendors current Tailwind, or to a stable release once one includes it). That's the highest-leverage single change; no oxfmt-side code needs to move.

I can send a PR if there's a preferred bump target — happy to take the direction.

**Comment by benface** [comment]
Closing as not-oxc — the root cause is in Tailwind's own `--value(...)` argument preprocessor (missing an `!arg.includes("(")` guard), which lands in oxfmt through two hops of vendoring: `tailwindcss` → `prettier-plugin-tailwindcss` (snapshots Tailwind's code at build time) → `oxfmt` (pins that plugin).

Fixed upstream in Tailwind 4.3.0 ([tailwindlabs/tailwindcss#19989](https://github.com/tailwindlabs/tailwindcss/pull/19989)); the plugin's committed lockfile pins Tailwind to 4.1.18 so rebuilds vendor the pre-guard code. PR to bump the pin: [tailwindlabs/prettier-plugin-tailwindcss#472](https://github.com/tailwindlabs/prettier-plugin-tailwindcss/pull/472). Once that merges and oxfmt picks up the new insider, this warning goes away with no oxc-side change needed.

Sorry for the noise.
